### PR TITLE
Add Tensorflow 2.12 tests for TPU VM.

### PR DESCRIPTION
**Pull request by gkroiz**
Copied files over from nightly tests, modified target versions, and made a few other smaller adjustments

### CODE REVIEW

File: tests/tensorflow/r2.12/mnist.libsonnet [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      '--tpu=$(KUBE_GOOGLE_CLOUD_TPU_ENDPOINTS)',
+    ],
+    flags+:: {
+      dataDir: '/tmp/mnist',
+      modelDir: '$(LOCAL_OUTPUT_DIR)',
+    },
+  },
+
+  configs: [
+    mnist + k80 + functional + mixins.Experimental,
```
Comment (chandrasekhard2) [inline tests/tensorflow/r2.12/mnist.libsonnet:99]: This is experimental test and can be removed for testing on main release.

**Review by chandrasekhard2 — APPROVED** [review]

File: tests/tensorflow/r2.12/mnist.libsonnet (hunk 1 above)
Comment (gkroiz) [reply tests/tensorflow/r2.12/mnist.libsonnet]: removed

**Comment by jiangjy1982** [comment]
Please populate the description of this PR: How the new files are generated? Did you copy them from somewhere? Or are they auto-generated from some meta-config?

**Review by jiangjy1982 — APPROVED** [review]

**Comment by will-cromar** [comment]
FYI the diff is empty. This PR doesn't add any tests:

```
Run # New files need to be added before git diff will recognize them.
From https://github.com/GoogleCloudPlatform/ml-testing-accelerators
 * branch            gen        -> FETCH_HEAD
 * [new branch]      gen        -> origin/gen
 ```

File: tests/tensorflow/r2.12/dlrm.libsonnet [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    paramsOverride+:: {
+      task+: {
+        model+: {
+          bottom_mlp: [512, 256, 128],
+          embedding_dim: 128,
+        },
+      },
+    },
+  },
+  local tpuVm = common.tpuVm,
```
Comment (aman2930) [inline tests/tensorflow/r2.12/dlrm.libsonnet:239]: tpu.Vm is not there in common.libsonnet.
Reference: https://github.com/gkroiz/ml-testing-accelerators/blob/5962332e531c7858cc68c9463668434b9c45ef38/tests/tensorflow/r2.11/common.libsonnet#L30

**Pull request merged**